app/Community/Commands/ConvertGameShortcodesToHubs.php

```diff
@@ -0,0 +1,253 @@
+    {
+        if ($this->option('undo')) {
+            $this->info("\nUndoing the migration of [hub=id] shortcodes back to [game=id].");
+
+            $this->undoForumMigration();
+            $this->undoMessagesMigration();
+        } else {
+            $this->info("\nStarting the migration of legacy hub [game=id] shortcodes to [hub=id].");
+
+            $this->migrateForumShortcodes();
+            $this->migrateMessageShortcodes();
+        }
+    }
+
+    private function migrateForumShortcodes(): void
+    {
+        // Get the total count of comments that need to be processed.
+        $totalComments = ForumTopicComment::where('body', 'like', '%[game=%')->count();
+
+        $progressBar = $this->output->createProgressBar($totalComments);
+        ForumTopicComment::where('body', 'like', '%[game=%')->chunkById(1000, function ($forumTopicComments) use ($progressBar) {
+            // Collect all game IDs to fetch hub mappings in bulk.
+            $gameIds = [];
+
+            /** @var ForumTopicComment $forumTopicComment */
+            foreach ($forumTopicComments as $forumTopicComment) {
+                preg_match_all('/\[game=(\d+)\]/', $forumTopicComment->body, $matches);
+                $gameIds = array_merge($gameIds, $matches[1]);
+            }
+
+            // Remove duplicates and fetch games that are legacy hubs.
+            $gameIds = array_unique($gameIds);
+            $legacyHubs = Game::whereIn('ID', $gameIds)
+                ->where('ConsoleID', 100)
+                ->get()
+                ->keyBy('ID');
+
+            // Fetch corresponding hub IDs for these legacy games.
+            $hubMappings = GameSet::where('type', GameSetType::Hub)
+                ->whereIn('game_id', $legacyHubs->pluck('ID'))
+                ->get()
+                ->keyBy('game_id');
+
+            // Process each comment.
+            /** @var ForumTopicComment $forumTopicComment */
+            foreach ($forumTopicComments as $forumTopicComment) {
+                $originalPayload = $forumTopicComment->body;
+                $updatedPayload = preg_replace_callback('/\[game=(\d+)\]/', function ($matches) use ($legacyHubs, $hubMappings) {
+                    $gameId = (int) $matches[1];
+
+                    // Only replace if it's a legacy hub and we have a mapping.
+                    if ($legacyHubs->has($gameId) && $hubMappings->has($gameId)) {
+                        return "[hub={$hubMappings->get($gameId)->id}]";
+                    }
+
+                    return $matches[0];
+                }, $forumTopicComment->body);
+
+                // Update the comment in the DB only if it has actually changed.
+                // Use `DB` so we don't change the `updated_at` value.
+                if ($originalPayload !== $updatedPayload) {
+                    $forumTopicComment->body = $updatedPayload;
+                    $forumTopicComment->timestamps = false;
+                    $forumTopicComment->save();
```

Comment:
`timestamps=false` didn't seem to work:
![image](https://github.com/user-attachments/assets/ac3f5c71-f28c-4553-8ac1-69e26abfd0d9)

Reply:
So strange - I didn't see it working on the original iteration and did see it working after I swapped over.

Reverted to the `DB::` calls in latest.